### PR TITLE
pre select destination and preferred route

### PR DESCRIPTION
**Pull request by sebastianscatularo**
Added `toTokenKey` and `preferredRouteName` params to `ui` object config to allow integrators to pre-select target token and mark a configured route as preferred

If the toTokenKey does not exist or is invalid, it will fail silent
If the preferred route does not exist or is invalid, it will fail silent

Sample config piece

```json
"defaultInputs": {
  "fromChain": "Ethereum",
  "toChain": "Arbitrum",
  "tokenKey": "USDCeth",
  "toTokenKey": "USDCarbitrum",
  "preferredRouteName": "ManualCCTP"
}
```

https://github.com/user-attachments/assets/9fb444f8-b4da-4868-9435-ec8f34b38634

### CODE REVIEW

File: wormhole-connect/src/views/v2/Bridge/index.tsx [hunk 1]
```diff
@@ -165,7 +167,11 @@ const Bridge = () => {
     if (validRoutes.length === 0) {
       setSelectedRoute('');
     } else {
-      const autoselectedRoute = route || validRoutes[0].route.name;
+      const preferredRoute = validRoutes.find(
+        (route) => route.route.name === preferredRouteName,
+      );
+      const autoselectedRoute =
+        route ?? validRoutes[0].route.name ?? preferredRoute?.route.name;
```
Comment (artursapek) [inline wormhole-connect/src/views/v2/Bridge/index.tsx]: Shouldn't the preferred route come before the `validRoutes[0].route.name`?

**Review by emreboga — DISMISSED** [review]
LGTM!

File: wormhole-connect/src/hooks/useComputeDestinationTokens.ts [hunk 2]
```diff
@@ -54,8 +55,15 @@ const useComputeDestinationTokens = (props: Props): ReturnProps => {
...
       setIsFetching(false);
 
-      if (destChain && supported.length === 1) {
-        if (!canceled) {
+      if (active && destChain && destToken !== '') {
+        // check if the pre selected destToken is supported
+        const isTokenSupported = supported.some((t) => t.key == destToken);
+        if (!isTokenSupported) {
+          // if not, clear the destToken
+          dispatch(setDestToken(''));
```
Comment (artursapek) [inline wormhole-connect/src/hooks/useComputeDestinationTokens.ts]: Why is this necessary? Can't we just let the UI show no quotes if there are no routes?

Comment (sebastianscatularo) [reply wormhole-connect/src/hooks/useComputeDestinationTokens.ts]: you're right, removed

File: wormhole-connect/src/views/v2/Bridge/index.tsx (hunk 1 above)
Comment (sebastianscatularo) [reply wormhole-connect/src/views/v2/Bridge/index.tsx]: yes, I've tried to optimize it, and that is why I've moved to the last position, but, you are right, updated

Comment (artursapek) [reply wormhole-connect/src/views/v2/Bridge/index.tsx]: wdym by "optimize it"?

**Review by artursapek — APPROVED** [review]
the diff in `wormhole-connect/src/hooks/useComputeDestinationTokens.ts` is pointless now but lgtm

**Comment by tsadovska** [comment]
Checked on Netlify https://wormhole-connect-mainnet.netlify.app/, `toTokenKey` and `preferredRouteName` configs work correctly. If token or route does not exist, then `toTokenKey` or `preferredRouteName` are ignored silently as expected.
